packages/features/ee/teams/lib/getUserAdminTeams.ts

```diff
@@ -0,0 +1,47 @@
+import { prisma } from "@calcom/prisma";
+import { MembershipRole } from "@calcom/prisma/enums";
+
+/** Get a user's team & orgs they are admins/owners of. Abstracted to a function to call in tRPC endpoint and SSR. */
+const getUserAdminTeams = async ({ userId, getUserInfo }: { userId: number; getUserInfo?: boolean }) => {
+  const teams = await prisma.team.findMany({
+    where: {
+      members: {
+        some: {
+          userId: userId,
+          role: { in: [MembershipRole.ADMIN, MembershipRole.OWNER] },
+        },
+      },
+    },
+    select: {
+      id: true,
+      name: true,
+      logo: true,
+    },
+  });
+
+  if (teams.length && getUserInfo) {
+    const user = await prisma.user.findUnique({
+      where: {
+        id: userId,
+      },
+      select: {
+        id: true,
+        name: true,
+        avatar: true,
+      },
+    });
+
+    if (user) {
+      const userObject = { id: user.id, name: user.name || "Nameless", logo: user?.avatar, isUser: true };
+      teams.unshift(userObject);
+    }
+  }
+
+  return teams;
+};
+
+export default getUserAdminTeams;
+
+export type UserAdminTeams =
+  | (Awaited<Promise<ReturnType<typeof getUserAdminTeams>>>[number] & { isUser?: boolean })[]
+  | [];
```

Comment:
Can be simplified a little bit. 

We dont need | [] from what i can see as even if there isnt any teams the typeof the array would still be 
`teams: {
    id: number;
    name: string;
    logo: string | null;
}[]
`
Just empty 

```suggestion
export type UserAdminTeams = (Awaited<ReturnType<typeof getUserAdminTeams>>[number] & { isUser?: boolean })[];

```

Reply:
This wasn't picking up the property `isUser` so I decided to change this. Line 6 of the same file